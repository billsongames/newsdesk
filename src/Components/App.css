.App {
  margin: auto;
  width: 1000px;
  position: static
}