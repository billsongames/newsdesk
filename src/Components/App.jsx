import React, {useState, useEffect, useContext} from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import { supabase } from '../api/api';

import './App.css';

import { DarkModeProvider } from '../context/DarkModeContext';

import Navbar from "./NavBar/NavBar"
import ArticleContainer from './ArticleContainer/ArticleContainer';
import Account from "./Account/Account"

function App() {

  const [userID, setUserID] = useState()
  const [articleCategory, setArticleCategory] = useState("general")
  const [search, setSearch] = useState("")
  const [sourceSelected, setSourceSelected] = useState("")

  const handleLogin = (response) => {
    console.log(response)
    setUserID(response.id)
  }

  const handleLogout = () => {
    window.FB.logout()
    setUserID()
  }

  const handleArticleCategoryChange = (newCategory) => {
    setSearch("")
    setArticleCategory(newCategory)
    setSourceSelected("")
  }

  const handleSearchRequest = (searchQuery) => {
    setArticleCategory("")
    setSearch(searchQuery)
  }

  const resetSource = () => {
    setSourceSelected("")

  }

  useEffect(() => {
    async function checkIfUserExists() {
      if (userID) {
        const { count, error } = await supabase
          .from('users')
          .select('user_id', {count: 'exact', head: true})
          .eq('user_id', userID)
  
          if (error) {
            console.log("error", error)
          } else {
          if (count === 0) {
            const { data, error } = await supabase
              .from('users')
              .insert([
                {
                  user_id: userID,
                  saved_articles: []
                }])

              if (error) {
                console.log("error", error)
              } else {
              console.log("insert success")   
              }
            }
          }
      }
    }
    checkIfUserExists()
    
  }, [userID]);


  return (
    <BrowserRouter>
      <div className="App">
        <DarkModeProvider>
          <Navbar onNavBarCategoryChange={handleArticleCategoryChange} onSearchSubmit={handleSearchRequest} userID={userID} onLogin={handleLogin} onLogout={handleLogout}/>
          <Routes>
            <Route path = "/" element={<ArticleContainer userID={userID} articleCategory={articleCategory} searchQuery={search} sourceSelected={sourceSelected}/>} />
            <Route path = "/saved-articles" element={<Account userID= {userID} />} />
          </Routes>
        </DarkModeProvider>


      </div>
    </BrowserRouter>
  );
}

export default App;
