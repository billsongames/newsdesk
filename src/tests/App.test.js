import { render } from '@testing-library/react';
import App from '../Components/App';

describe('Name of the group', () => {
  beforeEach(() => {
    const script = document.createElement('script');
    document.body.appendChild(script);
  })
  it('App renders correctly', () => {
    const { asFragment } =  render(<App />);
    
    expect(asFragment()).toMatchSnapshot()
  });
});
