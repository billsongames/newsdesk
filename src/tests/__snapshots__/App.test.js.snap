// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="navbar"
      >
        <div
          class="date-display"
        >
          <div
            class="date-display"
          >
            Wednesday 15 March 2023
          </div>
          <div
            class="dark-toggle"
          >
            <img
              alt="Dark Mode toggle"
              src="/lightswitch_off.png"
            />
          </div>
        </div>
        <div
          class="navbar-container"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              class="navbar-logo"
              src="/logo2_light.png"
            />
          </a>
          <div
            class="navbar-categories"
          >
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                General
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                World
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                National
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                Business
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                Technology
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                Entertainment
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                Sports
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                Science
              </button>
            </a>
            <a
              href="/"
            >
              <button
                class="navbar-button"
              >
                Health
              </button>
            </a>
          </div>
        </div>
        <div
          class="navbar-log-in-search-container"
        >
          <div
            class="navbar__facebook-login"
          >
            <span
              style="transition: opacity 0.5s; opacity: 0.6;"
            >
              <button
                class="my-facebook-button-class metro"
                type="button"
              >
                Login with Facebook
              </button>
            </span>
          </div>
          <form
            class="search-form"
          >
            <input
              class="search-form__input"
              placeholder="Search articles"
              type="text"
              value=""
            />
            <button
              class="search-form__button"
              data-testid="button"
              type="submit"
            >
              <img
                src="./search-icon-50px_light.png"
              />
            </button>
          </form>
        </div>
      </div>
      <div
        class="article-container"
      >
        <div>
          <div
            class="sidebar-container"
          >
            <div
              class="sidebar-filters-text"
            >
              Sort by Date:
              <button
                class="sidebar-button"
                data-testid="button1"
                disabled=""
              >
                Most Recent
              </button>
              <button
                class="sidebar-button"
                data-testid="button2"
              >
                Oldest
              </button>
            </div>
            <div
              class="sidebar-filters-text"
              data-testid="filter"
            >
              Filter by Source:
              <select>
                console.log()
                <option
                  value=""
                >
                  -All Sources-
                </option>
                <option
                  value="The Mirror"
                >
                  The Mirror
                </option>
                <option
                  value="Daily Mail"
                >
                  Daily Mail
                </option>
                <option
                  value="The Guardian"
                >
                  The Guardian
                </option>
                <option
                  value="Chronicle Live"
                >
                  Chronicle Live
                </option>
              </select>
            </div>
          </div>
        </div>
        <div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Peter Schmeichel identifies Man Utd's "odd one out" as he criticises transfer policy
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Peter Schmeichel launched a scathing attack on Manchester United's transfer business during the January window after they fell behind against Barcelona on Thursday night"
                    class="image"
                    src="https://i2-prod.mirror.co.uk/sport/article29307731.ece/ALTERNATES/s1200/0_Screenshot-2023-02-24-183152.jpg"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Peter Schmeichel launched a scathing attack on Manchester United's transfer business during the January window after they fell behind against Barcelona on Thursday night
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Peter Schmeichel launched a scathing attack on Manchester United's transfer business during the January window after they fell behind against Barcelona on Thursday night
Peter Schmeichel believes Erik ten Hag should have given an opportunity to a you... [3126 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.mirror.co.uk/sport/football/news/manutd-transfers-barcelona-tenhag-weghorst-29307406"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: The Mirror
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Man United fans shower 'top guy' Lisandro Martinez with praise after he stayed to sign autographs
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="United progressed to the last 16 of the Europa League with 4-3 aggregate win over Barcelona at Old Trafford. Martinez shone in the game at centre-back as Antony scored the winner."
                    class="image"
                    src="https://i.dailymail.co.uk/1s/2023/02/24/13/68040871-0-image-a-72_1677245696849.jpg"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    United progressed to the last 16 of the Europa League with 4-3 aggregate win over Barcelona at Old Trafford. Martinez shone in the game at centre-back as Antony scored the winner.
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Lisandro Martinez delighted Manchester United fans by staying late to sign autographs on Thursday.
United progressed to the last 16 of the Europa League with 4-3 aggregate win over Barcelona at Old Trafford.
Martinez shone in the game at centre-back ... [1805 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.dailymail.co.uk/sport/football/article-11789305/Man-United-fans-shower-guy-Lisandro-Martinez-praise-stayed-sign-autographs.html"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: Daily Mail
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Alejandro Garnacho is SLAMMED by Spanish media for his 'very poor taste' after mocking Pedri
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="The 18-year-old was a second half substitute in the match at Old Trafford on Thursday in a dramatic game that saw the Red Devils dump the Spanish giants out of the Europa League."
                    class="image"
                    src="https://i.dailymail.co.uk/1s/2023/02/24/13/68040175-0-image-a-60_1677243747363.jpg"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    The 18-year-old was a second half substitute in the match at Old Trafford on Thursday in a dramatic game that saw the Red Devils dump the Spanish giants out of the Europa League.
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Alejandro Garnacho has been slammed by the Spanish press following Manchester United's game against Barcelona.
The 18-year-old was a second half substitute in the match at Old Trafford on Thursday in a dramatic game that saw the Red Devils dump the S... [1739 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.dailymail.co.uk/sport/football/article-11789147/Alejandro-Garnacho-SLAMMED-Spanish-media-poor-taste-mocking-Pedri.html"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: Daily Mail
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Europa League draw, League Cup finals and Sellés in at Saints: football news - live
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Counting down to the weekend of football action. Join our writers for the latest news"
                    class="image"
                    src="https://i.guim.co.uk/img/media/a9ab5a57ffd7b108899e8ae9dcb3466d5c3e80f2/0_218_6534_3920/master/6534.jpg?width=1200&height=630&quality=85&auto=format&fit=crop&overlay-align=bottom%2Cleft&overlay-width=100p&overlay-base64=L2ltZy9zdGF0aWMvb3ZlcmxheXMvdGctbGl2ZS5wbmc&enable=upscale&s=5471925e040078c8c85ec256af9c203e"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Counting down to the weekend of football action. Join our writers for the latest news
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    11m ago 06.25 EST The Europa League Round of 16 draw, in which Arsenal have been paired with Sporting and Manchester United with Real Betis. Photograph: Denis Balibouse/Reuters
14m ago 06.22 EST Europa League Round of 16 draw FC Union Saint-Gilloise ... [5715 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.theguardian.com/football/live/2023/feb/24/league-cup-finals-europa-league-draw-and-selles-in-at-saints-football-news-live"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: The Guardian
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Europa League fixtures: Man Utd vs Real Betis, Arsenal vs Sporting Lisbon
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Manchester United will face more Spanish opposition in the form of Real Betis in the Europa League last-16, while Arsenal have been paired with Sporting Lisbon."
                    class="image"
                    src="https://i.dailymail.co.uk/1s/2023/02/24/11/68036921-0-image-a-27_1677238515090.jpg"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Manchester United will face more Spanish opposition in the form of Real Betis in the Europa League last-16, while Arsenal have been paired with Sporting Lisbon.
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Manchester United will face more Spanish opposition in the form of Real Betis in the Europa League last-16, while Arsenal have been paired with Sporting Lisbon.
Erik ten Hag's side overcame Barcelona 4-3 on aggregate in a thrilling play-off round tie... [3184 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.dailymail.co.uk/sport/football/article-11788829/Europa-League-draw-Manchester-United-play-Real-Betis-Arsenal-Sporting-Lisbon.html"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: Daily Mail
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Erik ten Hag reveals his half-time team talk that inspired comeback victory over Barcelona
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Robert Lewandowski gave the Spanish giants the lead in the first-half before Fred and Antony both got themselves on the scoresheet in the second-half to send United through to the last-16."
                    class="image"
                    src="https://i.dailymail.co.uk/1s/2023/02/24/08/68031335-0-image-a-18_1677225806454.jpg"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Robert Lewandowski gave the Spanish giants the lead in the first-half before Fred and Antony both got themselves on the scoresheet in the second-half to send United through to the last-16.
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Manchester United boss Erik ten Hag has revealed that he urged his players to believe in themselves at half-time of their 2-1 comeback victory over Barcelona in the Europa League.
Robert Lewandowski gave the Spanish giants the lead in the first-half ... [2603 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.dailymail.co.uk/sport/football/article-11788211/Erik-ten-Hag-reveals-half-time-team-talk-inspired-comeback-victory-Barcelona.html"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: Daily Mail
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Rio Ferdinand picks out Man Utd's two most important players from Barcelona win
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Former Manchester United defender Rio Ferdinand was full of praise for Antony and Fred after they both scored in the crucial Europa League play-off match against Barcelona"
                    class="image"
                    src="https://i2-prod.mirror.co.uk/incoming/article29269776.ece/ALTERNATES/s1200/0_Screenshot-2361.jpg"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Former Manchester United defender Rio Ferdinand was full of praise for Antony and Fred after they both scored in the crucial Europa League play-off match against Barcelona
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Former Manchester United defender Rio Ferdinand was full of praise for Antony and Fred after they both scored in the crucial Europa League play-off match against Barcelona
Rio Ferdinand has hailed Antony and Fred after the Brazilian duo both scored f... [2651 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.mirror.co.uk/sport/football/news/man-utd-rio-ferdinand-barcelona-29300088"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: The Mirror
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Eddie Howe reveals Kevin Keegan plan Newcastle will 'definitely arrange' even if 'difficult'
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Newcastle United head coach Eddie Howe reached out to Kevin Keegan before he even took the job and the Magpies boss will be inviting the club legend to speak to his players"
                    class="image"
                    src="https://i2-prod.chroniclelive.co.uk/sport/football/football-news/article26314040.ece/ALTERNATES/s1200/0_Untitled-design.png"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Newcastle United head coach Eddie Howe reached out to Kevin Keegan before he even took the job and the Magpies boss will be inviting the club legend to speak to his players
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Eddie Howe has vowed to 'definitely' arrange for Kevin Keegan to come in to speak to his Newcastle United players - even if it is not possible before the Carabao Cup final.
Keegan was one of the first people Howe phoned before the Newcastle boss form... [1783 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.chroniclelive.co.uk/sport/football/football-news/newcastle-eddie-howe-kevin-keegan-26313084"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: Chronicle Live
              </div>
              <div
                class="article-card-minor__time"
              >
                18 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Clinical finish from Antony takes Manchester United past Barcelona
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Second-half goals from Antony and Fred helped United beat Barcelona 2-1 to take them into the last 16 of the Europa League"
                    class="image"
                    src="https://i.guim.co.uk/img/media/82b09bb4197888f88edbf2e717f97e1f4f471af3/0_228_4640_2784/master/4640.jpg?width=1200&height=630&quality=85&auto=format&fit=crop&overlay-align=bottom%2Cleft&overlay-width=100p&overlay-base64=L2ltZy9zdGF0aWMvb3ZlcmxheXMvdGctZGVmYXVsdC5wbmc&enable=upscale&s=f660a0caf6530cf8d4d896f940af627c"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Second-half goals from Antony and Fred helped United beat Barcelona 2-1 to take them into the last 16 of the Europa League
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    It is probably fair to say that the jury has been out on Antony since Manchester United’s decision to pay €100m for him last summer. The most insistent criticism has been that he is too predictable. Defenders know that he wants to come in from the ri... [5726 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.theguardian.com/football/2023/feb/23/clinical-finish-from-antony-takes-manchester-united-past-barcelona"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: The Guardian
              </div>
              <div
                class="article-card-minor__time"
              >
                19 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="article-card-minor"
            >
              <div
                class="article-card-minor__title"
              >
                Bruno Fernandes criticised after 'stupid' decision gifts Barcelona a penalty against Man United
              </div>
              <div
                class="article-card-minor__display"
              >
                <div
                  class="article-card-minor__image"
                >
                  <img
                    alt="Former Manchester United goalkeeper Peter Schmeichel has criticised Bruno Fernandes for a stupid decision after the midfielder gifted Barcelona a penalty in their Europa League tie."
                    class="image"
                    src="https://i.dailymail.co.uk/1s/2023/02/23/21/68016619-0-image-a-38_1677187438310.jpg"
                  />
                </div>
                <div
                  class="article-card-minor__text"
                >
                  <div
                    class="article-card-minor__description"
                  >
                    Former Manchester United goalkeeper Peter Schmeichel has criticised Bruno Fernandes for a stupid decision after the midfielder gifted Barcelona a penalty in their Europa League tie.
                  </div>
                  <div
                    class="article-card-minor__content"
                  >
                    Former Manchester United goalkeeper Peter Schmeichel has criticised Bruno Fernandes for a stupid decision after the midfielder gifted Barcelona a penalty in their Europa League tie.
Fernandes was adjudged to have pulled back Alejandro Balde in the pe... [1594 chars] 
                    <a
                      class="article-card-minor__url"
                      href="https://www.dailymail.co.uk/sport/football/article-11786427/Bruno-Fernandes-criticised-stupid-decision-gifts-Barcelona-penalty-against-Man-United.html"
                      target="blank"
                    >
                      Read full article
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="article-card-minor__source"
              >
                Source: Daily Mail
              </div>
              <div
                class="article-card-minor__time"
              >
                19 days ago
              </div>
              <div
                class="article-buttons"
              >
                <div />
                <div
                  class="social-media"
                >
                  <div>
                    <button
                      class="article-save-button-login-prompt"
                      disabled=""
                    >
                      Log in to share and save articles
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
